ofo-argo-utils/db_logger.py

```diff
@@ -0,0 +1,127 @@
+            password=password
+        )
+        return conn
+    except Exception as e:
+        sys.stderr.write(f"Error connecting to database: {e}\n")
+        sys.exit(1)
+
+def log_datasets_initial(datasets, workflow_id):
+    conn = get_db_connection()
+    try:
+        cursor = conn.cursor()
+        for dataset in datasets:
+            cursor.execute(
+                """
+                INSERT INTO automate_metashape (dataset_name, workflow_id, status) 
+                VALUES (%s, %s, %s) 
+                ON CONFLICT (dataset_name, workflow_id) DO NOTHING
+                """,
+                (dataset, workflow_id, 'queued')
+            )
+        conn.commit()
+    except Exception as e:
+        sys.stderr.write(f"Error logging to database: {e}\n")
+        sys.exit(1)
+    finally:
+        if conn is not None:
+            conn.close()
+
+def log_dataset_start(dataset, workflow_id):
+    conn = get_db_connection()
+    try:
+        cursor = conn.cursor()
+        cursor.execute(
+            """
+            UPDATE automate_metashape 
+            SET status = 'processing', start_time = CURRENT_TIMESTAMP 
+            WHERE dataset_name = %s AND workflow_id = %s
+            """,
+            (dataset, workflow_id)
+        )
+        conn.commit()
+    except Exception as e:
+        sys.stderr.write(f"Error logging start time: {e}\n")
+        sys.exit(1)
+    finally:
+        if conn is not None:
+            conn.close()
+
+def log_dataset_completion(dataset, workflow_id, success):
+    conn = get_db_connection()
+    status = 'completed' if success else 'failed'
+    
+    try:
+        cursor = conn.cursor()
+        cursor.execute(
+            """
+            UPDATE automate_metashape 
+            SET status = %s, finish_time = CURRENT_TIMESTAMP 
+            WHERE dataset_name = %s AND workflow_id = %s
+            """,
+            (status, dataset, workflow_id)
+        )
+        conn.commit()
+    except Exception as e:
+        sys.stderr.write(f"Error logging completion: {e}\n")
+        sys.exit(1)
+    finally:
+        if conn is not None:
+            conn.close()
+
+def main():
+    parser = argparse.ArgumentParser(description='Database logging for Argo workflows')
+    
+    parser.add_argument('action', choices=['log-initial', 'log-start', 'log-completion'],
+                      help='Logging action to perform')
+    
+    parser.add_argument('--workflow-id', required=True,
+                      help='Workflow ID to use in database records')
+    
+    parser.add_argument('--datasets-json',
+                      help='JSON array of dataset names for initial logging')
+    parser.add_argument('--dataset',
+                      help='Dataset name for start/completion logging')
+    parser.add_argument('--success',
+                      help='Success status (true/false) for completion logging')
+    
+    args = parser.parse_args()
+    
+    if args.action == 'log-initial':
```

Comment:
Not necessary for this PR, but I often like to decouple the CLI parsing logic and business logic of the script into two separate functions. An example is [here](https://github.com/open-forest-observatory/geograypher/blob/main/geograypher/entrypoints/aggregate_images.py). My reasoning is then the function that performs the actual logic can easily be reused, for example in a notebook or a driver script that executes many calls.